### PR TITLE
Fix label "Web only" 

### LINKED ISSUE #9959: [BOOM-3364] Bandeau exclusivité web
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-3364).

- _**Reporter:**_ marion_francois
- _**Created at:**_ Thu, 29 Jun 2017 17:58:43 +0200

<p>Le bandeau exclusivité web masque les déclinaisons</p>
        
* How to reproduce the issue ?

Create or update a product and check &quot;Web Only&quot; in Product Option. See FO.

### PR DESCRIPTION
**Pull request by YeLnatSs**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fix on issue when product is sold online only.
| Type?         | bug fix 
| Category?     | FO 
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | Fixes #9959 
| How to test?  | In BO on product page, check "web only" in options. Update FO and you'll be able to see "Online only" label on product

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12872)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -292,3 +292,5 @@ nb-configuration.xml
 ### Release creation ###
 tools/build/releases/*
 !tools/build/releases/.gitkeep
+admin-dev/themes/new-theme/public/module.bundle.js
```
Comment (PierreRambaud) [inline .gitignore]: Can you give us more information about why ignoring this file twice?
And public files must be add in git :)

File: admin-dev/themes/new-theme/.webpack/common.js [hunk 2]
```diff
@@ -139,7 +139,7 @@ module.exports = {
               loader: 'css-loader',
               options: {
                 minimize: true,
-                sourceMap: true,
+                sourceMap: true
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/.webpack/common.js]: Why? This is a rule from our eslint config.

File: admin-dev/themes/new-theme/.webpack/dev.js [hunk 3]
```diff
@@ -17,7 +17,6 @@ function devConfig() {
       },
     }
   );
-
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/.webpack/dev.js]: same here

File: admin-dev/themes/new-theme/.webpack/prod.js [hunk 4]
```diff
@@ -31,7 +31,6 @@ function prodConfig(analyze) {
   );
 
   return prod;
-
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/.webpack/prod.js]: 👍

File: admin-dev/themes/new-theme/.webpack/dev.js (hunk 3 above)
Comment (YeLnatSs) [reply admin-dev/themes/new-theme/.webpack/dev.js]: I changed it

File: admin-dev/themes/new-theme/.webpack/common.js (hunk 2 above)
Comment (YeLnatSs) [reply admin-dev/themes/new-theme/.webpack/common.js]: I changed it

File: .gitignore (hunk 1 above)
Comment (YeLnatSs) [reply .gitignore]: Didn't know about that one. I changed it

**Comment by colinegin** [comment]
#9959

File: admin-dev/themes/new-theme/.webpack/prod.js [hunk 5]
```diff
@@ -31,7 +31,7 @@ function prodConfig(analyze) {
...
+
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/.webpack/prod.js]: useless spaces.

File: admin-dev/themes/new-theme/.webpack/dev.js [hunk 6]
```diff
@@ -17,8 +17,8 @@ function devConfig() {
...
   return dev;
+
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/.webpack/dev.js]: No need space here

Comment (YeLnatSs) [reply admin-dev/themes/new-theme/.webpack/dev.js]: space removed

File: admin-dev/themes/new-theme/.webpack/prod.js (hunk 5 above)
Comment (YeLnatSs) [reply admin-dev/themes/new-theme/.webpack/prod.js]: space removed

File: admin-dev/themes/new-theme/.webpack/dev.js [hunk 7]
```diff
@@ -17,7 +17,7 @@ function devConfig() {
       },
     }
   );
-
+
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/.webpack/dev.js]: Useless space here

Comment (YeLnatSs) [reply admin-dev/themes/new-theme/.webpack/dev.js]: I removed it

**Comment by PierreRambaud** [comment]
I don't understand why js files changed whereas you didn't change anything in javascript :thinking:

**Comment by LouiseBonnard** [comment]
Hi, it seems that there is no string to check in this PR. Can you confirm?

**Comment by eternoendless** [comment]
Rebased, squashed, removed merge commits and wrong files.

@YeLnatSs In the future please make sure NOT to commit merge commits, and rebase as necessary.

**Review by eternoendless — DISMISSED** [review]

**Comment by marionf** [comment]
@YeLnatSs 

It's not fixed on categories pages

capture d'écran_1221

It should be like this

1706_web_only_OK


And it's too high on the product pages

capture d'écran_1223

It should be like this

capture d'écran_1224

**Comment by YeLnatSs** [comment]
@marionf I fixed it. You can check it now

**Comment by marionf** [comment]
Thank you @YeLnatSs I am waiting for code review because I see a lot of conflicting files and don't know if it's normal

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Something is wrong with your commit:
+4,083 −25,604
474 files changed

**Comment by YeLnatSs** [comment]
Hello @marionf can you check this PR again?

File: themes/classic/_dev/css/components/products.scss [hunk 8]
```diff
@@ -520,6 +520,15 @@ li.product-flag {
         margin-top: 3.75rem;
     }
 }
+#product {
```
Comment (PierreRambaud) [inline themes/classic/_dev/css/components/products.scss]: `#product` was already spotted on line 1.

**Comment by marionf** [comment]
@YeLnatSs it exceeds the image on the right and it's too hight

capture d'écran_1237

Can you do it like below ?

capture d'écran_1238

**Comment by YeLnatSs** [comment]
> @YeLnatSs it exceeds the image on the right and it's too hight
> 
> capture d'écran_1237
> 
> Can you do it like below ?
> 
> capture d'écran_1238

@marionf Okay, for the label that exceeds, it was done according to the opinion of @TristanLDD but I can changed if needed.

**Comment by marionf** [comment]
@YeLnatSs In the home page it's still not correctly placed

capture d'écran_1239

**Comment by marionf** [comment]
> @marionf Okay, for the label that exceeds, it was done according to the opinion of @TristanLDD but I can changed if needed.

Let's wait for @TristanLDD :)

**Comment by YeLnatSs** [comment]
@marionf I saw with @TristanLDD. You can check it now

**Comment by marionf** [comment]
All good, thank you @YeLnatSs

**Comment by PierreRambaud** [comment]
PR needs to be rebased

**Comment by Quetzacoalt91** [comment]
PR rebased

**Comment by Quetzacoalt91** [comment]
On my shop, there is an issue when I also have a flag "On sale" on the same product.

Capture d’écran du 2019-03-27 18-07-52

**Comment by marionf** [comment]
Nice catch @Quetzacoalt91 

I tried with all labels and it's looks like this

capture d'écran_1246

**Comment by YeLnatSs** [comment]
@marionf can you try it again?

**Comment by marionf** [comment]
@YeLnatSs Could you see with @TristanLDD to improve it

capture d'écran_1256

The "new" label is now like this

capture d'écran_1255

Before it was like this

capture d'écran_1257

@TristanLDD what do you think ?

**Comment by TristanLDD** [comment]
@marion the expected behavior for the labels is the following: 

- Every label should appear on the top left of product thumbnail (Except the "web only" label which goes on bottom right) 
- If there are two (or more) labels on top left, the second goes under the first one (with a 10px spacing) , etc 

I'll check that with @YeLnatSs today ;)

**Comment by marionf** [comment]
@TristanLDD Is it ok for you like this ?

capture d'écran_1271

capture d'écran_1272

capture d'écran_1273

**Comment by YeLnatSs** [comment]
@marionf I checked and fixed with @TristanLDD about it. You can review it

File: themes/classic/_dev/css/components/products.scss [hunk 9]
```diff
@@ -681,4 +680,4 @@ li.product-flag {
       padding: 1rem;
     }
   }
-}
+}
```
Comment (PierreRambaud) [inline themes/classic/_dev/css/components/products.scss]: You remove the last line of this file.+

File: themes/classic/_dev/js/components/product-select.js [hunk 10]
```diff
@@ -32,6 +32,7 @@ export default class ProductSelect {
     let $arrows =   $('.js-modal-arrows');
     let $thumbnails = $('.js-modal-product-images');
     let $onsale =  $('.on-sale');
+    let $new = $('.new');
```
Comment (PierreRambaud) [inline themes/classic/_dev/js/components/product-select.js:36]: Why create this $new variable if you're not using it line 53 :thinking:

Comment (PierreRambaud) [reply themes/classic/_dev/js/components/product-select.js:36]: Do the same with `$('.pack')`

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
see comments

File: themes/classic/_dev/js/components/product-select.js (hunk 10 above)
Comment (YeLnatSs) [reply themes/classic/_dev/js/components/product-select.js:36]: I forgot to remove it.  I add a new variable for pack.

File: themes/classic/_dev/css/components/products.scss (hunk 9 above)
Comment (YeLnatSs) [reply themes/classic/_dev/css/components/products.scss]: Just add it

File: themes/classic/_dev/js/components/product-select.js [hunk 11]
```diff
@@ -32,7 +32,7 @@ export default class ProductSelect {
     let $arrows =   $('.js-modal-arrows');
     let $thumbnails = $('.js-modal-product-images');
     let $onsale =  $('.on-sale');
-    let $new = $('.new');
```
Comment (PierreRambaud) [inline themes/classic/_dev/js/components/product-select.js]: I didn't say you need to remove it. just to use it. 
It's not consistent to have $pack and no $new, or $new and no $pack. You must use both variables or not.

Comment (YeLnatSs) [reply themes/classic/_dev/js/components/product-select.js]: I wasn't using it.
Why was $new wasn't create before ?
I'll create it for the one which already exist.

Comment (YeLnatSs) [reply themes/classic/_dev/js/components/product-select.js]: I created it

File: themes/classic/_dev/js/components/product-select.js [hunk 12]
```diff
@@ -49,8 +51,14 @@ export default class ProductSelect {
     });
 
     if($onsale.length && $('#product').length){
-      $('.new').css('top',$onsale.height() + FLAG_MARGIN);
+      $new.css('top',$onsale.height() + 1 * FLAG_MARGIN);
```
Comment (PierreRambaud) [inline themes/classic/_dev/js/components/product-select.js]:
```suggestion
      $new.css('top', $onsale.height() + 1 * FLAG_MARGIN);
```

File: themes/classic/_dev/js/components/product-select.js [hunk 13]
```diff
@@ -49,8 +51,14 @@ export default class ProductSelect {
...
+      $pack.css('top',$onsale.height() + 1 * FLAG_MARGIN);
```
Comment (PierreRambaud) [inline themes/classic/_dev/js/components/product-select.js]:
```suggestion
      $pack.css('top', $onsale.height() + 1 * FLAG_MARGIN);
```

File: themes/classic/_dev/js/components/product-select.js [hunk 14]
```diff
@@ -49,8 +51,14 @@ export default class ProductSelect {
...
     }
+
+    if($new.length && $('#product').length) {
+      $pack.css('top',$onsale.height() + 5 * FLAG_MARGIN);
```
Comment (PierreRambaud) [inline themes/classic/_dev/js/components/product-select.js]:
```suggestion
      $pack.css('top', $onsale.height() + 5 * FLAG_MARGIN);
```

File: themes/classic/_dev/js/components/product-select.js [hunk 15]
```diff
@@ -49,8 +51,14 @@ export default class ProductSelect {
     });
 
     if($onsale.length && $('#product').length){
-      $('.new').css('top',$onsale.height() + FLAG_MARGIN);
+      $new.css('top',$onsale.height() + 1 * FLAG_MARGIN);
+      $pack.css('top',$onsale.height() + 1 * FLAG_MARGIN);
     }
+
+    if($new.length && $('#product').length) {
```
Comment (PierreRambaud) [inline themes/classic/_dev/js/components/product-select.js]:
```suggestion
    if ($new.length && $('#product').length) {
```

Comment (YeLnatSs) [reply themes/classic/_dev/js/components/product-select.js]: done

File: themes/classic/_dev/js/components/product-select.js (hunk 14 above)
Comment (YeLnatSs) [reply themes/classic/_dev/js/components/product-select.js]: done

File: themes/classic/_dev/js/components/product-select.js (hunk 13 above)
Comment (YeLnatSs) [reply themes/classic/_dev/js/components/product-select.js]: done

File: themes/classic/_dev/js/components/product-select.js (hunk 12 above)
Comment (YeLnatSs) [reply themes/classic/_dev/js/components/product-select.js]: done

File: themes/classic/_dev/css/components/featuredproducts.scss [hunk 16]
```diff
@@ -191,4 +191,4 @@ $product-description-height: 70px;
       }
     }
   }
-}
+}
```
Comment (PierreRambaud) [inline themes/classic/_dev/css/components/featuredproducts.scss]: `No newline at the end `

Comment (YeLnatSs) [reply themes/classic/_dev/css/components/featuredproducts.scss]: added

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @YeLnatSs

**Comment by marionf** [comment]
Seems, it's not completely fixed

Between -3,00€ and "New", we should have 10px spacing

Capture d’écran 2019-04-12 à 15 49 44


Here, "Pack" and "new" are still superimposed

Capture d’écran 2019-04-12 à 15 50 23

**Pull request merged**